## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :x: [more...](#9567d70f-6230-4271-88dd-f8d7417b1f27) |
<a name="9567d70f-6230-4271-88dd-f8d7417b1f27"></a>
### 15112-7344811/mint-pools.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s5:30454
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 8 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 44 seconds
(4/15) Running aws-sdk-ruby tests ... done in 7 seconds
(5/15) Running awscli tests ... done in 56 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 29 seconds
(8/15) Running minio-dotnet tests ... done in 6 minutes and 20 seconds
(9/15) Running minio-go tests ... FAILED in 5 minutes and 10 seconds
panic: Get "http://minio-k8s5:30454/minio-go-test-jizbi1cpdiy20tj6/50mzs6zkkszexulxgn5bloqxpilfy4": context deadline exceeded

goroutine 1 [running]:
main.crcMatches({0x955e00, 0xc000373a00}, 0xc5a7128a)
	/mint/run/core/minio-go/main.go:315 +0x12e
main.testGetObjectRanges()
	/mint/run/core/minio-go/main.go:11409 +0xe45
main.main()
	/mint/run/core/minio-go/main.go:12125 +0x2af
(9/15) Running minio-java tests ... done in 35 seconds
(10/15) Running minio-js tests ... done in 1 minutes and 6 seconds
(11/15) Running minio-py tests ... done in 1 minutes and 36 seconds
(12/15) Running s3cmd tests ... done in 26 seconds
(13/15) Running s3select tests ... done in 5 seconds
(14/15) Running versioning tests ... done in 3 minutes and 5 seconds

Executed 14 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

--- a/cmd/admin-heal-ops.go
+++ b/cmd/admin-heal-ops.go
@@ -91,8 +91,11 @@ type allHealState struct {
 	sync.RWMutex
 
 	// map of heal path to heal sequence
-	healSeqMap     map[string]*healSequence  // Indexed by endpoint
-	healLocalDisks map[Endpoint]bool         // healing in progress
+	healSeqMap map[string]*healSequence // Indexed by endpoint
+	// keep track of the healing status of disks in the memory
+	//   false: the disk needs to be healed but no healing routine is started
+	//    true: the disk is currently healing
+	healLocalDisks map[Endpoint]bool
 	healStatus     map[string]healingTracker // Indexed by disk ID
 }
 
@@ -158,6 +161,8 @@ func (ahs *allHealState) getLocalHealingDisks() map[string]madmin.HealingDisk {
 	return dst
 }
 
+// getHealLocalDiskEndpoints() returns the list of disks that need
+// to be healed but there is no healing routine in progress on them.
 func (ahs *allHealState) getHealLocalDiskEndpoints() Endpoints {
 	ahs.RLock()
 	defer ahs.RUnlock()

